src/engine/IndexScan.cpp

```diff
@@ -212,15 +216,42 @@ void IndexScan::computePOSfreeO(ResultTable* result) const {
 }
 
 // _____________________________________________________________________________
-size_t IndexScan::computeSizeEstimate() const {
+size_t IndexScan::computeSizeEstimate() {
   if (_executionContext) {
     // Should always be in this branch. Else is only for test cases.
 
     // We have to do a simple scan anyway so might as well do it now
     if (getResultWidth() == 1) {
-      return getResult()->size();
+      RuntimeInformation& runtimeInfo = getRuntimeInfo();
+      bool firstRun = runtimeInfo.getTime() == 0;
+      size_t size = getResult()->size();
+      // When a cached result is loaded but the runtimeInfo for that
+      // Operation object was already computed the old values are
+      // used (instead of the time for the cache access. This breaks
+      // the output of the operation time for any operation that has
+      // a single column scan as its child, as that operation itself
+      // would measure the time of the cache access for the result
+      // of that scan, while the scan itself would report the time
+      // required for the actual scan that is done in here. As both
+      // times are interestin we add the time for the initial scan
+      // as a detail to the RuntimeInformation and then reset the
+      // runtime information time to ensure the resulting runtime
+      // information trees operation runtimes are actually correct.
+      if (firstRun) {
+        runtimeInfo.addDetail("IntialScanTime",
```

Comment:
@floriankramer  ok this doesn't work as is. The problem is that in `getResult()` the `newResult->_runtimeInfo` is set via copying and only after that this detail is added. Then later during the actual query we retrieve the old copy which doesn't have the detail in it.

Reply:
Maybe we should only do the `runtimeInfo.setTime(0)` below and then when taking a `RuntimeInformation` from the cache always save the original times in the details?

Reply:
That's a good idea.